### PR TITLE
Nested Components not updating the child component

### PR DESCRIPTION
**Pull request by carvinlo**
Nested Components not updating the child component

# New Pull Request

For more information, see the `CONTRIBUTING` guide.

## Prerequisites

Please make sure you can check the following boxes:

- [x ] I have read the **CONTRIBUTING** document
- [x ] My code follows the code style of this project
- [x ] All new and existing tests passed

### Contribution Type

What types of changes does your code introduce? Put an `x` in all the boxes that apply:

- [x ] Bug fix (non-breaking change which fixes an issue, please reference the issue id)
- [ ] New feature (non-breaking change which adds functionality, make sure to open an associated issue first)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] I have updated the documentation accordingly
- [ ] I have added tests to cover my changes

### Description

[Place a meaningful description here]

### Remarks

[Optionally place any follow-up comments, remarks, observations, or notes here for future reference]

### CODE REVIEW

File: src/framework/piral-core/src/components/wrapComponent.tsx [hunk 1]
```diff
@@ -31,15 +31,15 @@ function wrapForeignComponent<T>(
...
+    const id = React.useMemo(() => (portalIdBase++).toString(26), [props.children]);
     // router added for backwards compatibility
     const context = React.useMemo(
       () => ({ publicPath: navigation.publicPath, navigation, router: navigation.router }),
       [],
     );
     const innerProps = React.useMemo(() => ({ ...props, ...captured }), [props]);
 
-    React.useEffect(() => () => destroyPortal(id), none);
+    React.useEffect(() => () => destroyPortal(id), [props.children]);
```
Comment (FlorianRappl) [inline src/framework/piral-core/src/components/wrapComponent.tsx:42]: Sure you want to destroy the portal then? I don't think that makes sense here.

File: src/framework/piral-core/src/components/wrapComponent.tsx [hunk 2]
```diff
@@ -31,15 +31,15 @@ function wrapForeignComponent<T>(
 ) {
   return React.memo((props: T) => {
     const { destroyPortal, navigation } = useGlobalStateContext();
-    const id = React.useMemo(() => (portalIdBase++).toString(26), none);
+    const id = React.useMemo(() => (portalIdBase++).toString(26), [props.children]);
```
Comment (FlorianRappl) [inline src/framework/piral-core/src/components/wrapComponent.tsx:34]: I don't see why a change of props.children should have an impact on the id of the portal.

Maybe let's first have a MWE to reproduce the issue and then let's see what the root cause is / how to improve it.

Comment (carvinlo) [reply src/framework/piral-core/src/components/wrapComponent.tsx:34]: The situation is a bit complicated, but it does affect the update of components, resulting in abnormal rendering. I will show you this MWE later.

Comment (FlorianRappl) [reply src/framework/piral-core/src/components/wrapComponent.tsx:34]: Sure - thanks for the efforts!

Comment (FlorianRappl) [reply src/framework/piral-core/src/components/wrapComponent.tsx:34]: Any update on this? 1.5.3 is out and 1.5.4 is in the making.

Comment (carvinlo) [reply src/framework/piral-core/src/components/wrapComponent.tsx:34]: Yep. It is in processing, I'm using the develop branch to make the MWE.

Comment (carvinlo) [reply src/framework/piral-core/src/components/wrapComponent.tsx:34]: I need more time to review the details of the project, so I will close it for now and will reopen it when there is progress. It's most likely not the `piral-core`'s problem, and sorry for this.

Comment (FlorianRappl) [reply src/framework/piral-core/src/components/wrapComponent.tsx:34]: All good - just reopen / open an issue if we have a reproducible or can pin it down to Piral in any way.

**Pull request closed without merging**